CMakeLists.txt

```diff
@@ -12,18 +12,37 @@ find_package(sensor_msgs REQUIRED)
 find_package(geometry_msgs REQUIRED)
 find_package(rosbag2_interfaces REQUIRED)
 find_package(yaml-cpp REQUIRED)
+find_package(rclcpp_lifecycle REQUIRED)
+find_package(lifecycle_msgs REQUIRED)
 
 ###########
 ## Build ##
 ###########
 
-ament_export_dependencies(rclcpp geometry_msgs sensor_msgs rosbag2_interfaces)
+# ament_export_dependencies(rclcpp geometry_msgs sensor_msgs rosbag2_interfaces lifecycle_msgs rclcpp_lifecycle)
 
 set(UNIFIED_TELEOP_AMENT_TARGET_DEPENDENCIES
-  rclcpp geometry_msgs sensor_msgs rosbag2_interfaces
+  rclcpp geometry_msgs sensor_msgs rosbag2_interfaces lifecycle_msgs rclcpp_lifecycle
 )
+
+# Create a library
+add_library(${PROJECT_NAME} src/controller.cpp include/unified_teleop/param_helpers.hpp include/unified_teleop/pnt_stmp_modifiers.hpp)
```

Comment:
Don't use ${PROJECT_NAME}, just name the library.

Reply:
Okay overall, I think I did what you asked me to do for the CMakeLists file but if not let me know